docs/src/apimanual.md

```diff
@@ -412,11 +412,210 @@ primal_variable_result = MOI.get(optimizer, MOI.VariablePrimal(), x)
+function of a model. These modifications can be grouped into two categories:
+modifications which replace the set of function of a constraint with a new set
+or function; and modifications which change, in-place, a component of a
+function.
+
+In the following, we detail the various ways this can be
+achieved. Readers should note that some solvers will not support problem
+modification.
+
+### Replacements
+
+First, we discuss how to replace the set or function of a constraint with a new
+instance of the same type.
+
+#### The set of a constraint
+
+Given a constraint of type `F`-in-`S` (see [Constraints by function-set pairs](@ref)
+ above for an explanation), we can modify parameters (but not the type) of the
+ set `S` by replacing it with a new instance of the same type. For example,
+ given the variable bound ``x \\le 1``:
+```julia
+c = addconstraint(m, SingleVariable(x), LessThan(1.0))
+```
+we can modify the set so that the bound now ``x \\le 2`` as follows:
+```julia
+set!(m, ConstraintSet(), c, LessThan(2.0))
+```
+where `m` is our [`ModelLike`](@ref) model. However, the following will fail as
+the new set (`GreaterThan`) is of a different type to the original set
+(`LessThan`):
+```julia
+set!(m, ConstraintSet(), c, GreaterThan(2.0))  # errors
+```
+If our constraint is an affine inequality, then this corresponds to modifying
+the right-hand side of a constraint in linear programming.
+
+In some special cases, solvers may support efficiently changing the set of a
+constraint (for example, from [`LessThan`](@ref) to [`GreaterThan`](@ref)). For
+these cases, MathOptInterface provides the [`transform!`](@ref) method. For
+example, instead of the error we observed above, the following will work:
+```julia
+c2 = transform!(m, c, GreaterThan(1.0))
+```
+The [`transform!`](@ref) function returns a new constraint index, and the old
+constraint index (i.e., `c`) is no longer valid:
+```julia
+isvalid(m, c)   # false
+isvalid(m, c2)  # true
+```
+Also note that [`transform!`](@ref) cannot be called with a set of the same
+type; [`set!`](@ref) should be used instead.
+
+#### The function of a constraint
+
+Given a constraint of type `F`-in-`S` (see [Constraints by function-set pairs](@ref)
+above for an explanation), it is also  possible to modify the function of type
+`F` by replacing it with a new instance of the same type. For example, given the
+variable bound ``x \\le 1``:
+```julia
+c = addconstraint(m, SingleVariable(x), LessThan(1.0))
+```
+we can modify the function so that the bound now ``y \\le 1`` as follows:
+```julia
+set!(m, ConstraintFunction(), c, SingleVariable(y))
+```
+where `m` is our [`ModelLike`](@ref) model. However, the following will fail as
+the new function is of a different type to the original function:
+```julia
+set!(m, ConstraintFunction(), c,
+    ScalarAffineFunction([ScalarAffineTerm(1.0, x)], 0.0)
+)
+```
+
+### In-place Modifications
```

Comment:
Nit: Inconsistent capitalization of words in section titles. (Also "Problem Modification" at the top).